src/box/sql/build.c

```diff
@@ -2008,6 +2009,61 @@ sql_drop_table(struct Parse *parse_context)
 	sqlSrcListDelete(db, table_name_list);
 }
 
+/** Generate code to change the definition of tuple constraint. */
+static void
+vdbe_emit_alter_constraint(struct Parse *parse, struct space_def *def)
+{
+	struct region *region = &parse->region;
+	uint32_t opts_size;
+	const char *opts = sql_encode_table_opts(region, def, &opts_size);
+	if (opts == NULL) {
+		parse->is_aborted = true;
+		return;
+	}
+	uint32_t format_size;
+	const char *format = sql_encode_table(region, def, &format_size);
+	if (format == NULL) {
+		parse->is_aborted = true;
+		return;
+	}
+	char *raw = sqlDbMallocRaw(parse->db, opts_size + format_size);
+	if (raw == NULL) {
+		parse->is_aborted = true;
+		return;
+	}
+	memcpy(raw, opts, opts_size);
+	memcpy(raw + opts_size, format, format_size);
+	struct Vdbe *v = sqlGetVdbe(parse);
+	assert(v != NULL);
+	int id_reg = sqlGetTempReg(parse);
+	sqlVdbeAddOp2(v, OP_Integer, def->id, id_reg);
+	int key_reg = sqlGetTempReg(parse);
+	sqlVdbeAddOp3(v, OP_MakeRecord, id_reg, 1, key_reg);
+	int format_reg = sqlGetTempReg(parse);
+	sqlVdbeAddOp4(v, OP_Blob, format_size, format_reg, SQL_SUBTYPE_MSGPACK,
+		      raw + opts_size, P4_STATIC);
+	int opts_reg = sqlGetTempReg(parse);
+	sqlVdbeAddOp4(v, OP_Blob, opts_size, opts_reg, SQL_SUBTYPE_MSGPACK, raw,
+		      P4_DYNAMIC);
+	uint32_t *upd_cols = sqlDbMallocRaw(parse->db, sizeof(*upd_cols) * 4);
+	if (upd_cols == NULL)
+		return;
```

Comment:
Should there be `is_aborted = true`?

Reply:
Hi! Thank you for the review! I'm sorry, I reworked this whole patch.

This part I dropped.